### PR TITLE
chore: encourage changing appname parameter from default

### PR DESCRIPTION
**Pull request by lazysoundsystem**
Refs: RW-839

The main change here is in the copy functionality, which swaps 'apidoc' for 'REPLACE-THIS-WITH-YOUR-EMAIL-ADDRESS' - suggestions welcome for a better string here. If we make it something that no one would be using now, we could get the API to reject any requests with that parameter.

Wondering too whether we could use the same string in the search converter for anonymous users instead of `rwint-0`

The other major change is the paragraph in `_meta/api-use.html`. I don't know if this is sufficiently loud to make a difference. 

The other changes are small but noisy: changing the sppname parameter in post requests from part of the url to part of the json. I think it might make things more clear, but I'm in two minds.

Suggestions welcome!

### CODE REVIEW

**Review by orakili — COMMENTED** [review]
I don't think it's a good idea to move the appname from the query parameters to the payload for the POST requests because that will not appear in the logs anymore in that case.

Also not sure about mentioning the username as appname. Maybe another good suggestion rather than the email address is the domain name or app name.

**Comment by lazysoundsystem** [comment]
Good points - perhaps would be good to reject api calls with an appname of 'REPLACE-WITH-A-DOMAIN-OR-APP-NAME'.

Also we could alter the search converter to use the same text.

The warning is still not very prominent - we could check how many requests come in without an appname parameter and see if we can identify those, so we'd be able to make it obligatory going forward.

File: js/site.js [hunk 1]
```diff
@@ -109,16 +109,20 @@
       return response.json();
     })
     .then(function(json) {
+      console.log('options', options);
```
Comment (orakili) [inline js/site.js]: Leftover?

**Review by orakili — COMMENTED** [review]
@lazysoundsystem I'm wondering about reversing the logic for the `appname` in the API doc: replace all instances of `apidoc` with `REPLACE-WITH-A-DOMAIN-OR-APP-NAME` so it's clearly visible to the person reading the doc and internally, when clicking the "Try it out", replace that text with `apidoc` in the URL used to query the API and replace `apidoc` in the result from the API.

This way the reader would always see `REPLACE-WITH-A-DOMAIN-OR-APP-NAME` and all the text they would copy whether using the button or by selection would contain that string.

I feel that may be more robust and visible. What do you think?

**Comment by lazysoundsystem** [comment]
Good suggestion, feels more robust.

Still open to better strings to use than 'REPLACE-WITH-DOMAIN-OR-APP-NAME', but it's clear and definitely won't be in use already, so we can still block that new string in the API, if we want to, before merging this PR.

**Review by orakili — APPROVED** [review]
Looks good, thanks.

I don't think we need to block anything for now. Let's see the logs in some weeks to have an idea of how many people use the API without changing the appname.

**Pull request merged**